src/handler_error.rs

```diff
@@ -41,125 +41,171 @@ pub async fn handle_rejection(rej: Rejection) -> Result<impl Reply, Rejection> {
     //                10200: transaction rejected
     match rej.find::<ApiError>() {
         None => Err(rej),
-        Some(err) => {
-            let e = match err {
-                ApiError::UnsupportedFieldPresent(field_name) => {
-                    invalid_input_unsupported_field_error(Some(field_name.to_string()))
-                }
-                ApiError::SubAccountNotImplemented => {
-                    invalid_input_unsupported_field_error(Some("sub_account".to_string()))
-                }
-                ApiError::RequiredFieldMissing(name) => {
-                    invalid_input_missing_field_error(Some(name.to_string()))
-                }
-                ApiError::InvalidAccountAddress(addr) => {
-                    invalid_input_invalid_value_or_identifier_error(
-                        Some("account address".to_string()),
-                        None,
-                        Some(addr.clone()),
-                        Some("invalid format".to_string()),
-                    )
-                }
-                ApiError::InvalidCurrency => invalid_input_invalid_value_or_identifier_error(
+        Some(err) => Ok(match err {
+            ApiError::UnsupportedFieldPresent(field_name) => reply::with_status(
+                reply::json(&invalid_input_unsupported_field_error(Some(field_name.to_string()))),
+                StatusCode::BAD_REQUEST,
+            ),
+            ApiError::SubAccountNotImplemented => reply::with_status(
+                reply::json(&invalid_input_unsupported_field_error(Some(
+                    "sub_account".to_string(),
+                ))),
+                StatusCode::BAD_REQUEST,
+            ),
+            ApiError::RequiredFieldMissing(name) => reply::with_status(
+                reply::json(&invalid_input_missing_field_error(Some(name.to_string()))),
+                StatusCode::BAD_REQUEST,
+            ),
+            ApiError::InvalidAccountAddress(addr) => reply::with_status(
```

Comment:
The examples in the documentation https://www.rosetta-api.org/docs/AccountApi.html#500---error seem to show error 500 for this error, but also all the other errors that I've checked (by going through examples).

Not sure why they chose that, since BAD_REQUEST does make sense in this case, but seeing that they did choose it, shouldn't we follow that choice?

Reply:
Hm I never noticed that... They really don't seem to care about HTTP. I'll check/ask their forum if they have an official opinion.

Reply:
Ok. Let's wait with merging this until there you've done that.

Reply:
https://community.rosetta-api.org/t/http-status-codes/736

Reply:
@abizjak I guess we aren't going to get a response to that complex question. So what do we do?

Given the time that I've already spent and the fact that I consider it correct, I would prefer merging this as it is. But I would also be fine with changing it to all 500...

Reply:
Can we put a short section in the README explaining the choice of the more precise error codes, as opposed to 500 which the examples show?

With that, I think we can merge this.